crates/fluvio/src/producer/mod.rs

```diff
@@ -352,8 +419,9 @@ impl TopicProducer {
                 config,
                 topic,
                 spu_pool,
-                producer_pool,
-                record_accumulator,
+                producer_pool: Arc::new(RwLock::new(producer_pool)),
+                record_accumulator: Arc::new(record_accumulator),
+                metrics: metrics.clone(),
             }),
             #[cfg(feature = "smartengine")]
```

Comment:
seems need unit/integration to tests producers

Reply:
How to handle SpuPool in this case? Create a mock?

Reply:
Could build a fake SPU handler that serve respond to socket similar to integration tests in SPU

Reply:
As talked in Company Meeting: it's better to merge it now for a new release and create a new pull request with unit tests.

Reply:
Issue: https://github.com/infinyon/fluvio/issues/4093

Reply:
Why is better to merge now? Since this significant change and effect core path,  there should be enough test coverage.  Simple e2e is not sufficient.

Reply:
I don't remember the reason. 

Anyway, I tried to write integration tests on fluvio package, but it's needed to import `fluvio-spu`, and seems that `fluvio-spu` package is not very friendly to be imported.

Also tried to write them on `fluvio-spu` but `ProducerTopic` and other structs needs are private or protected (pub(crate)).

What do you think? Which side should we test it?

By the way, I noticed that there are some tests in `fluvio-test` using TopicProducer: 
- longevity 
- producer 
- producer_fail

Reply:
Probably don't want to to use SPU directly in the client.  All this is way to test behavior changes.  This could be mocking SPU response (may need to change internal interface to make it easier).

Reply:
Hmm. The only solution I came to mind is to use traits + generics. 
But this will be a big break change for connectors! As `TopicProduce` will be changed to `TopicProduce<S>`.

I added a new commit these changes + a test, what do you think?